programs/account-compression/src/instructions/append_leaves.rs

```diff
@@ -111,16 +126,34 @@ pub fn process_append_leaves_to_merkle_trees<'a, 'b, 'c: 'info, 'info>(
             paths.push(path);
         }
 
-        #[cfg(target_os = "solana")]
-        light_heap::GLOBAL_ALLOCATOR.log_total_heap("append_leaves: after changelog event");
-
         changelog_events.push(ChangelogEvent::V1(ChangelogEventV1 {
             id: merkle_tree_pubkey.to_bytes(),
             paths,
             seq: first_sequence_number as u64,
             index: merkle_tree.changelog[first_changelog_index].index as u32,
         }));
+
+        leaves_start += leaves_end;
+
+        if leaves_start == BATCH_SIZE {
+            emit_event(&ctx, changelog_events)?;
+
+            leaves_start = 0;
+            changelog_events = Vec::with_capacity(BATCH_SIZE);
```

Comment:
we allocate  heap memory again for every batch we should make every batch heap neutral.